modules/src/core/ics04_channel/events.rs

```diff
@@ -1006,28 +1118,40 @@ mod tests {
             counterparty_channel_id: Some("counterparty_test_channel".parse().unwrap()),
         };
         let mut abci_events = vec![];
-        let open_init = OpenInit::from(attributes.clone());
+        let open_init = OpenInit::try_from(attributes.clone()).unwrap();
```

Comment:
I wonder if this could be refactored into extracting values from an `&Attributes` reference rather than cloning all of the attributes and then picking some of the values from the clone.

Reply:
Only one field in `Atributes` is movable so using a reference won't make much difference since will still have to clone all the other fields to use them